rezervo/integrations/fsc/schema.py

```diff
@@ -0,0 +1,224 @@
+
+
+class BookingsResponse(FscResponse):
+    data: Optional[List[BookingData]] = None
+
+
+class Country(BaseModel):
+    id: int
+    name: str
+    alpha2: str
+
+
+class ShippingAddress(BaseModel):
+    postalCode: str
+    city: str
+    street: str
+    careOf: str
+    country: Country
+
+
+class MobilePhone(BaseModel):
+    number: str
+    countryCode: int
+
+
+class CustomerType(BaseModel):
+    id: int
+    name: str
+
+
+class ProfileImage(BaseModel):
+    id: int
+
+
+class Consent(BaseModel):
+    id: int
+    name: str
+
+
+class UserDetails(BaseModel):
+    id: int
+    firstName: str
+    lastName: str
+    sex: str
+    ssn: Optional[str] = None
+    birthDate: str
+    shippingAddress: ShippingAddress
+    billingAddress: Optional[
+        dict
+    ] = None  # You can create a separate BillingAddress class if needed
+    email: str
+    mobilePhone: MobilePhone
+    businessUnit: BusinessUnit  # Reusing existing BusinessUnit schema
+    customerType: CustomerType
+    customerTypeEndDate: Optional[str] = None
+    customerNumber: str
+    cardNumber: str
+    acceptedBookingTerms: bool
+    acceptedSubscriptionTerms: bool
+    acceptedRegistrationTerms: str
+    profileImage: ProfileImage
+    benefitStatus: Optional[Union[int, str]] = None  # Type based on actual usage
+    memberJoinDate: str
+    allowMassSendEmail: bool
+    allowMassSendMail: bool
+    allowMassSendSms: bool
+    consents: List[Consent]
+    temporary: Optional[Union[int, str]] = None  # Type based on actual usage
+    lastPasswordChangedTime: str
+
+
+class UserDetailsResponse(FscResponse):
+    data: UserDetails
+    auth: Dict[str, int]
+
+
+def session_state_from_fsc(booking_type: BookingType) -> SessionState:
+    match booking_type:
+        case BookingType.GROUP_ACTIVITY:
+            return SessionState.BOOKED
+        case BookingType.WAITING_LIST:
+            return SessionState.WAITLIST
+    return SessionState.UNKNOWN
+
+
+def to_local_date_str(date: str) -> str:
+    return date.replace("Z", "")[:-4]
+
+
+def rezervo_class_from_fsc_class(fsc_class: FscClass) -> RezervoClass:
+    return RezervoClass(
+        integration=IntegrationIdentifier.FSC,
+        id=fsc_class.id,
+        name=fsc_class.groupActivityProduct.name,
+        activityId=fsc_class.groupActivityProduct.id,
+        from_field=to_local_date_str(fsc_class.duration.start),
+        to=to_local_date_str(fsc_class.duration.end),
+        instructors=[RezervoInstructor(name=s.name) for s in fsc_class.instructors],
+        studio=RezervoStudio(
+            id=fsc_class.businessUnit.id,
+            name=fsc_class.businessUnit.name,
+        ),
+        userStatus=None,
+        bookable=datetime.fromisoformat(to_local_date_str(fsc_class.bookableEarliest))
+        < datetime.now()
+        < datetime.fromisoformat(to_local_date_str(fsc_class.bookableLatest)),
+        bookingOpensAt=to_local_date_str(fsc_class.bookableEarliest),
```

Comment:
Looks like this is UTC, so the crontab will be wrong

Reply:
fixed